### PR TITLE
feat: Any parsable URL will be added into the Draft body

### PR DESCRIPTION
**Pull request by adrien-coye**

### CODE REVIEW

File: Mail/Views/New Message/ComposeMessageView.swift [hunk 1]
```diff
@@ -118,9 +118,10 @@ struct ComposeMessageView: View {
         mailboxManager: MailboxManager,
         messageReply: MessageReply? = nil,
         attachments: [Attachable] = [],
-        textAttachment: TextAttachable? = nil
+        textAttachments: [TextAttachable] = []
```
Comment (adrien-coye) [inline Mail/Views/New Message/ComposeMessageView.swift:121]: Compose message view can now accept an arbitrary collection of `TextAttachable` not just one.

File: MailShareExtension/WeblocToTextAttachment.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+// import InfomaniakCore
```
Comment (PhilippeWeidmann) [inline MailShareExtension/WeblocToTextAttachment.swift:21]: Commented code

File: MailCore/Cache/Attachments/AttachmentsManagerWorker/AttachmentsManagerWorker.swift [hunk 3]
```diff
@@ -384,30 +395,62 @@ extension AttachmentsManagerWorker: AttachmentsManagerWorkable {
...
 
+    private func anyUsableTitle(in textAttachments: [TextAttachment]) -> String {
+        textAttachments.first { textAttachment in
+            guard let title = textAttachment.title,
+                  !title.isEmpty else {
+                return false
+            }
+
+            return true
+        }?.title ?? ""
```
Comment (PhilippeWeidmann) [inline MailCore/Cache/Attachments/AttachmentsManagerWorker/AttachmentsManagerWorker.swift:425]: Maybe it's juste me but I find this more readable than a big trailing closure with guard ?
```suggestion
        textAttachments.first { $0.title?.isEmpty == false }?.title ?? ""
```

Comment (adrien-coye) [reply MailCore/Cache/Attachments/AttachmentsManagerWorker/AttachmentsManagerWorker.swift:425]: To me longer can be easier to read, as less things are processed per line, but that's fine with me.

File: MailShareExtension/WeblocToTextAttachment.swift (hunk 2 above)
Comment (adrien-coye) [reply MailShareExtension/WeblocToTextAttachment.swift:21]: good catch

**Review by PhilippeWeidmann — APPROVED** [review]

**Pull request merged**